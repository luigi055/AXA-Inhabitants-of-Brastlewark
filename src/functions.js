// @flow
export const gnomesAPI = "https://raw.githubusercontent.com/rrafols/mobile_test/master/data.json";
export function convertColorsToHex(color: string) {
  const colorName = color.toLowerCase();
  let hexColor: "";
  switch (colorName) {
    case "red":
      hexColor = "#d63031";
      break;
    case "pink":
      hexColor = "#fd79a8";
      break;
    case "green":
      hexColor = "#00b894";
      break;
    case "black":
      hexColor = "#2d3436";
      break;
    case "gray":
      hexColor = "#636e72";
      break;
    default:
      hexColor = "#000";
      break;
  }
  return hexColor;
}

export function filterGnomesBy(array, orderByFilter) {
  if (!Array.isArray(array)) {
    return null;
  }

  if (orderByFilter.toLowerCase() === "Oldest".toLowerCase()) {
    return array.sort((a, b) => b.age - a.age);
  } else if (orderByFilter.toLowerCase() === "Youngest".toLowerCase()) {
    return array.sort((a, b) => a.age - b.age);
  } else if (orderByFilter.toLowerCase() === "Heaviest".toLowerCase()) {
    return array.sort((a, b) => b.weight - a.weight);
  } else if (orderByFilter.toLowerCase() === "lightest".toLowerCase()) {
    return array.sort((a, b) => a.weight - b.weight);
  } else if (orderByFilter.toLowerCase() === "Most Popular".toLowerCase()) {
    return array.sort((a, b) => b.friends.length - a.friends.length);
  } else if (orderByFilter.toLowerCase() === "Less Popular".toLowerCase()) {
    return array.sort((a, b) => a.friends.length - b.friends.length);
  } else if (orderByFilter.toLowerCase() === "Reset".toLowerCase()) {
    return array.sort((a, b) => a.id - b.id);
  } else if (orderByFilter.toLowerCase() === "hair color".toLowerCase()) {
    /* eslint-disable */
    // Temporally while working on this
    alert("WORKING ON IT");
    /* eslint-enable */
  }

  return array;
}

export function setCookie(name, value, exdays) {
  const now = new Date();
  now.setTime(now.getTime() + exdays * 24 * 60 * 60 * 1000);
  const expires = `expires=${now.toUTCString()}`;
  document.cookie = `${name}=${value};${expires};path=/`;
}

// borrowed from w3schools
export function getCookie(cname) {
  const name = `${cname}=`;
  const decodedCookie = decodeURIComponent(document.cookie);
  const ca = decodedCookie.split(";");
  for (let i = 0; i < ca.length; i + 1) {
    let c = ca[i];
    while (c.charAt(0) === " ") {
      c = c.substring(1);
    }
    if (c.indexOf(name) === 0) {
      return c.substring(name.length, c.length);
    }
  }
  return "";
}
