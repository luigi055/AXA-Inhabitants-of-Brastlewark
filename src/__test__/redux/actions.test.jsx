// @flow
import moxios from "moxios";
import thunk from "redux-thunk";
import { gnomesAPI } from "./../../functions";
import { fakeGnomesData, localStorage } from "./../seed";
import {
  REQUEST_GNOMES,
  SEARCH_TERM,
  FILTER_BY_JOB,
  ORDER_BY_FILTER
} from "./../../redux/actions/types";
import {
  getGnomes,
  getSearchText,
  filterByJob,
  updateOrderBy,
  fetchGnomes
} from "./../../redux/actions/actions";

// emulate localStorage API
window.localStorage = localStorage;

// Tetsing all Actions Creator
describe("Testing Actions Creators", () => {
  beforeEach(() => {
    // pass custom axios instance
    moxios.install();
  });

  afterEach(() => {
    moxios.uninstall();
  });
  test("fetchGnomes", (done: Function) => {
    const dispatchMock = jest.fn();
    // Mocking axios
    moxios.withMock(() => {
      // Invoking fetchGnomes with dispaych mock
      fetchGnomes()(dispatchMock);
      moxios.wait(() => {
        const request = moxios.requests.mostRecent();
        // emulate the api get request
        request
          .respondWith({
            status: 200,
            response: { Brastlewark: fakeGnomesData }
          })
          .then(url => {
            expect(request.url).toEqual(gnomesAPI);
            expect(dispatchMock).toBeCalledWith(getGnomes(fakeGnomesData));
            done();
          })
          .catch(err => done(err));
      });
    });
  });

  describe("Update Search State Action Creator", () => {
    it('should generate "big gnome" in searchTerm State', () => {
      const searchGnome = "big gnome";
      const action = {
        type: SEARCH_TERM,
        payload: searchGnome
      };

      const searchActionCreator = getSearchText(searchGnome);
      expect(searchActionCreator).toEqual(action);
      expect(searchActionCreator).toMatchSnapshot();
    });
  });

  describe("Update filterByJob Action Creator", () => {
    it('should generate "Mason" in filterBy State', () => {
      const filterByJobValue = "Mason";
      const action = {
        type: FILTER_BY_JOB,
        payload: filterByJobValue
      };

      const filterByJobActionCreator = filterByJob(filterByJobValue);
      expect(filterByJobActionCreator).toEqual(action);
      expect(filterByJobActionCreator).toMatchSnapshot();
    });
  });
  describe("Update orderBy Action Creator", () => {
    it('should generate "Mason" in filterBy State', () => {
      const orderByValue = "Mason";
      const action = {
        type: ORDER_BY_FILTER,
        payload: orderByValue
      };

      const orderByActionCreator = updateOrderBy(orderByValue);
      expect(orderByActionCreator).toEqual(action);
      expect(orderByActionCreator).toMatchSnapshot();
    });
  });
});
