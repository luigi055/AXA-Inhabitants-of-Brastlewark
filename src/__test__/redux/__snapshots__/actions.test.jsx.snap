// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Actions Creators Update Search State Action Creator should generate "big gnome" in searchTerm State 1`] = `
Object {
  "payload": "big gnome",
  "type": "SEARCH_TERM",
}
`;

exports[`Testing Actions Creators Update filterByJob Action Creator should generate "Mason" in filterBy State 1`] = `
Object {
  "payload": "Mason",
  "type": "FILTER_BY_JOB",
}
`;

exports[`Testing Actions Creators Update orderBy Action Creator should generate "Mason" in filterBy State 1`] = `
Object {
  "payload": "Mason",
  "type": "ORDER_BY_FILTER",
}
`;
